packages/@aws-cdk/aws-vpcv2-alpha/lib/util.ts

```diff
@@ -0,0 +1,434 @@
+/*eslint no-bitwise: ["error", { "allow": ["~", "|", "<<", "&"] }] */
+
+import { ISubnet, SubnetType } from 'aws-cdk-lib/aws-ec2';
+
+/**
+ * Turn an arbitrary string into one that can be used as a CloudFormation identifier by stripping special characters
+ *
+ * (At the moment, no efforts are taken to prevent collisions, but we can add that later when it becomes necessary).
+ */
+export function slugify(x: string): string {
+  return x.replace(/[^a-zA-Z0-9]/g, '');
+}
+
+/**
+ * The default names for every subnet type
+ */
+export function defaultSubnetName(type: SubnetType) {
+  switch (type) {
+    case SubnetType.PUBLIC: return 'Public';
+    case SubnetType.PRIVATE_WITH_NAT:
+    case SubnetType.PRIVATE_WITH_EGRESS:
+    case SubnetType.PRIVATE:
+      return 'Private';
+    case SubnetType.PRIVATE_ISOLATED:
+    case SubnetType.ISOLATED:
+      return 'Isolated';
+  }
+}
+
+/**
+ * Return a subnet name from its construct ID
+ *
+ * All subnet names look like NAME <> "Subnet" <> INDEX
+ */
+export function subnetGroupNameFromConstructId(subnet: ISubnet) {
+  return subnet.node.id.replace(/Subnet\d+$/, '');
+}
+
+/**
+ * Make the subnet construct ID from a name and number
+ */
+export function subnetId(name: string, i: number) {
+  return `${name}Subnet${i + 1}`;
+}
+
+/**
+ * Generate the list of numbers of [0..n)
+ */
+export function range(n: number): number[] {
+  const ret: number[] = [];
+  for (let i = 0; i < n; i++) {
+    ret.push(i);
+  }
+  return ret;
+}
+
+/**
+ * Return the union of table IDs from all selected subnets
+ */
+export function allRouteTableIds(subnets: ISubnet[]): string[] {
+  const ret = new Set<string>();
+  for (const subnet of subnets) {
+    if (subnet.routeTable && subnet.routeTable.routeTableId) {
+      ret.add(subnet.routeTable.routeTableId);
+    }
+  }
+  return Array.from(ret);
+}
+
+export function flatten<A>(xs: A[][]): A[] {
+  return Array.prototype.concat.apply([], xs);
+}
+
+/**
+ * NetworkUtils contains helpers to work with network constructs (subnets/ranges)
+ */
+export class NetworkUtils {
+
+  /**
+   * Validates an IPv4 address string.
+   *
+   * @param ipAddress The IPv4 address string to be validated.
+   * @returns True if the string is a valid IPv4 address, false otherwise.
+   * Validates an IPv4 string
+   *
+   * returns true of the string contains 4 numbers between 0-255 delimited by
+   * a `.` character
+   */
+  public static validIp(ipAddress: string): boolean {
+    const octets = ipAddress.split('.');
+    if (octets.length !== 4) {
+      return false;
+    }
+    return octets.map((octet: string) => parseInt(octet, 10)).
+      filter((octet: number) => octet >= 0 && octet <= 255).length === 4;
```

Comment:
You already know that the length is 4, so this can be simplified a bit:

```suggestion
    return octets.map((octet: string) => parseInt(octet, 10)).
      every((octet: number) => octet >= 0 && octet <= 255;
```